ValveResourceFormat/Utils/UnexpectedMagicException.cs

```diff
@@ -28,5 +28,21 @@ public UnexpectedMagicException(string message, string magic, string nameofMagic
             Magic = magic;
             MagicNameof = nameofMagic;
         }
+
+        public static void ThrowIfNotEqual(int expectedMagic, int actualMagic, string nameofMagic)
```

Comment:
Can we use CallerArgumentExpression? I haven't tried it but I think that should work.

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/caller-argument-expression

Reply:
Seems nice. Did a quick test without editing the `Message`:
```
ValveResourceFormat.Utils.UnexpectedMagicException: Assertion failed for variable 'param.Id == 252': 255
```